back/spec/services/slug_service_spec.rb

```diff
@@ -1,48 +1,65 @@
 require 'rails_helper'
 
 describe SlugService do
-  let(:service) { SlugService.new }
+  let(:service) { described_class.new }
+
+  # We currently don't use this method directly, and we don't create project slugs in bulk, but it's a
+  # good example of how the method works.
+  # We do call this method from UserSlugService, and we have specs to cover that in UserSlugServiceSpec.
   describe 'generate_slugs' do
```

Comment:
I added these tests (of `generate_slugs` method) to explain and demonstrate how the related method works, but it feels a little strange to illustrate with anything other than a collection of users, since we currently only ever use this method, via a call from `UserSlugService`, when creating slugs for a collection of users.

More evidence that the related method should actually be moved int `UserSlugService`?

Reply:
In general, the code looks really nice!
I think it's totally fine to have `generate_slugs` in `SlugService` as it's not related directly to users.

Also, it seems the statement "We currently don't use this method directly" is not completely true as `generate_slugs` is used in `UserSlugService`. But it's not that important, so feel free to leave it as it is now.